### PR TITLE
Explicitly pass the team_id given to match through to cert and sigh

### PR DESCRIPTION
**Pull request by mfurtak**
Addresses https://github.com/fastlane/fastlane/issues/3996 and adds some tests

### CODE REVIEW

**Comment by asfalcone** [comment]
LGTM 👍

**Comment by mfurtak** [comment]
@KrauseFx please let me know if this is not how you would expect to solve this 👍

File: match/spec/generator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      allow(Sigh::Manager).to receive(:start).and_return("fake_path")
+
+      params = {
+        app_identifier: 'app_identifier',
+        prov_type: :development,
+        workspace: 'workspace',
+        username: 'username',
+        team_id: 'team_id'
+      }
+      Match::Generator.generate_provisioning_profile(params: params, prov_type: :development, certificate_id: 'fake_cert_id')
```
Comment (i2amsam) [inline match/spec/generator_spec.rb]: Nice!

File: match/spec/generator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        force: true,
+        cert_id: 'fake_cert_id',
+        provisioning_name: 'match Development app_identifier',
+        ignore_profiles_with_different_name: true,
+        team_id: 'team_id'
+      })
+
+      # This is the important part. We need to see the right configuration come through
+      # for sigh
+      expect(Sigh).to receive(:config=).with(a_configuration_matching(config))
```
Comment (KrauseFx) [inline match/spec/generator_spec.rb]: so beautiful

File: match/spec/spec_helper.rb [hunk 3]
```diff
@@ -9,3 +9,9 @@ module SpecHelper
 end
 
 WebMock.disable_net_connect!(allow: 'coveralls.io')
+
+RSpec::Matchers.define :a_configuration_matching do |expected|
```
Comment (KrauseFx) [inline match/spec/spec_helper.rb]: Awesome

**Comment by KrauseFx** [comment]
Perfect implementation @mfurtak, and great tests!!

**Pull request merged**